src/main/java/com/jcraft/jsch/Session.java

```diff
@@ -2544,11 +2604,14 @@ public int setPortForwardingR(String conf) throws JSchException {
    */
   public Channel getStreamForwarder(String host, int port) throws JSchException {
     ChannelDirectTCPIP channel = new ChannelDirectTCPIP();
-    channel.init();
-    this.addChannel(channel);
-    channel.setHost(host);
-    channel.setPort(port);
-    return channel;
+    if (this.addChannel(channel)) {
+      channel.init();
+      channel.setHost(host);
+      channel.setPort(port);
+      return channel;
+    } else {
+      throw new JSchException("Unable to add channel");
```

Comment:
I added this to throw an exception in case the session's `disconnect()` sequence has started and the channel cannot be added.